app/code/Magento/Catalog/Model/Product/Option/Validator/Select.php

```diff
@@ -27,6 +27,22 @@ protected function checkAllValuesRemoved($values)
     }
 
     /**
+     * Unset values marked for deleting
+     *
+     * @param $values
+     * @return $this
+     */
+    protected function cleanValues(&$values)
```

Comment:
New method should be declared as a private if there is no strong reasons to make it protected.
Passing values by reference is strongly deprecated in Magento 2. Please rewrite method and return filtered values explicitly. 
Also more descriptive method name (e.g. `filterDeletedValues`) will increase readability of the code.